parser.ts

```diff
@@ -185,6 +185,21 @@ export function traverseExpr(c : TreeCursor, s : string) : Expr<null> {
         tag: "id",
         name: "self"
       };
+    case "ArrayExpression":
+    case "TupleExpression":
+      let elements: Expr<null>[] = [];
+      c.firstChild();
+      c.nextSibling();
+      while (s.substring(c.from, c.to).trim() !== "]") {
```

Comment:
is this case for tuples? like `(1,2,3)` ? I'm a bit confused as why we are looking for `]` then? shouldn't that be `)`?

Reply:
That's a bug. The reason is that I wanted to reuse the logic between TupleExpression and ArrayExpression. I have separated this logic to make it clear and fixed the bug.